### DISCUSSION TITLE
Snap Card (outline + BG transparente)

### DISCUSSION THREAD

**Discussion by annefranca** [Telefonica/mistica-design#1343] [Proposals]
### Goal

Avaliar nova proposta para aplicação do componente Snap Card para ser adicionado no Mística.

### Context

Pensando em escalabilidade, solicitamos a aplicação do Snap Card com borda usando opacidade + background transparente. Dessa forma, vamos ter um componente consistente e reutilizável para todas as marcas.

### Impact

Adicionar essa lógica de opacidade e transparência traria facilidade na aplicação e deixaríamos de lado a necessidade de sempre ter que adicionar um novo token de cor para suprir as necessidades de uso.

### Discussion Questions

ND

### Additional considerations

Segue a mesma lógica da discussão do componente Chips.
https://github.com/Telefonica/mistica-design/discussions/1342

### Do you have a visual proposal?

https://www.figma.com/file/Xao9LLbt8NloneegKfAmKK/Snap-Card-(outline-%2B-BG-transparente)?type=design&node-id=0%3A1&mode=design&t=FF0ANo3f1jB6cPVB-1

**Comment by aweell** [comment]
Hi @annefranca,

The proposal of transparent background is only for inverse contexts? How the card should behave over default or over alternative?

**Reply by annefranca** [reply]
Olá @aweell 

Para aplicação em contextos diversos. No exemplo de aplicação do arquivo do Figma estamos aplicando sobre o backgroundBrandSecondary (vivoPurpleLight90 / #751AA3), mas poderia ser aplicado em qualquer outro contexto, como mostra a simulação. Lembrando que não é apenas o BG do Snap Card, mas também a opacidade na borda.


<img width="813" alt="Captura de Tela 2023-07-04 às 15 23 10" src="https://github.com/Telefonica/mistica-design/assets/127137234/e294dd01-1da2-44fe-97c1-ce8b52f4aa5e">

**Comment by yceballost** [comment]
What would happen to the current inverse cards? Would they be replaced by this transparent style?

I mean this
https://tinyurl.com/23xbjpxg

**Reply by annefranca** [reply]
@yceballost pensamos em ser uma outra opção sem sobrescrever o theme context atual, pois estamos utilizando elementos com opacidade, como na borda e subtítulo.

https://www.figma.com/file/Xao9LLbt8NloneegKfAmKK/Snap-Card-(outline-%2B-BG-transparente)?type=design&node-id=0%3A1&mode=design&t=HeKNP0pV6RLyooXF-1

**Reply by yceballost** [reply]
Hi! after evaluating it with the team we think the following:

1. We believe that the coexistence between "card inverse over inverse" and "transparent" can cause interaction problems, the user could understand that the dark one is not clickable. Here we should have data that this coexistence of cards works correctly, so far you have not provided such data.
2. It seems a very specific need for that screen. We have not seen more uses for this, so we do not consider it as something global.
3. We do not know what it brings to the user, other than the possible interaction problems mentioned in point 1.
4. We believe that it could be a problem to open so many styles to the cards, this could cause some places they are used with white backgrounds... others transparent... and in a product ecosystem, this could become a problem. How does a designer know when to use one style or another?

That said, for the moment we do not consider it part of the system, if in the future we unlock this kind of doubt we would have no problem in adding it to Mística. For the moment and being that screen your development, you could mount it in a Vivo Extended.

---

On the other hand, I leave the workaround that could fit if this comes some time to Mística.
The cards could have an `isTransparent` prop as currently has the `isInverse` prop. It would be necessary to evaluate if this would come in if there should be a prop that joins both cases... but this for the moment is food for the future.

**Reply by annefranca** [reply]
Obrigada @yceballost 👍 

Vamos avaliar com a equipe e o mais breve possível voltamos aqui com um direcionamento.

@BrunoBiagioniNeto alguma consideração?

**Reply by annefranca** [reply]
@aweell @AnaMontes11 
Segue a proposta atualizada: [Playroom](https://mistica-web.vercel.app/playroom#?code=N4Igxg9gJgpiBcIA8AFATjAbgSxgdwBUIIAbAZwAIB6APgB0A7RpAIQgA8KAHAQyimwMA5gF5gAJgAMAX3oMKFJAWwAXEjHE10ELvArYyBNDwZleGBiqRVlajXOtt2DgEowzEU9kwwAMjwBPCABXFX0yAEkGHzQyGApMHjRsExUROhBYADMeYJIVDIosvJIAdWwoFQALOQVWDm4%2BAWEATTEARgA2WUYFOoBxZKgKNAg8Mg7pCkgSYIBbUzEAFimhHi4xKR75PoGhiJUYOenSeYYAZV4GZamAK2CyFWwsgPOYEiz08BhLGDQM2q7PpKGDsFRLGgEKpHeIANSSKUsegAIjAcnkrDZQeDAbtrIMKgcjrjgQSoETjjwSNghBd3p8MmAfod-iAKPdHs9XvSvkzfqySXjzioeGAANYUMyimBiGSCoEgsHtGjCniHFFo3L5awEbHK3pAupUVXi%2BV1KLUhjxKVMr5JEIMKAAg2GuomiU2mXAAAc21dCotghgZtdSAEmBd-qBjwC6jEwEjUaBACMIGhYGgXHxsA89GQxYJ4bEAHSp9N-LMCbDFwQCMBqtMAGkTSb6ZYzegy7S4nDIpAqGWbO1bKbTGYAwqQ03oZmmyKWx38hyPdtJpC3-SH-fVOLx%2BIJRMB2uIpnvmkIABpfLrO4cruq6sGacdVbBcMg67GaDdRxwcLcKlQ4YAUagZWgB1juiG1hgcGLb4vshxzPKSBkhSFBUjSdIfLyzJ-IUHJPC8bw4YyeECj%2BSDupKvC2sAco-g%2BeoqiK6oUAAgmATyJIcUCfkq0HGiKpqUdRnpiL6EGwSBijhox0YqLGXoJneI7thW2a5pKBYMEW87qZm2bVrW2D1ioTbyYaBmdiA3a9v2TogMu94UAZk4kNOJweSWkCWKMJCcdxaowFAzn3smopikIozBI67mebOJbJsYjq%2BGMYWtmulkUDJdROI0%2B7CB0J4FeeV5dp0t4uYoj4qM%2Br7vvxdW5YoVBODJ1jAfJMEMJacGqcCQmRShVBoUhKFjUcGHUrSJEMt8-IEQ8RHcqRC0slVoZibRXoMQNeK1cqqpsai6LaliAmUUNIn7Yo%2BVngebRHieEHbdKyx7SOYbeNlCgxnGwAqdVBmVjmZB5jpekLuWhlVsWvkikGaAZWpi5oDZdmSg5g6-W2aPxejXlztDGYo1GWW3bsLU7qVB4dN0LVusJHo7fTfrVYoES%2BQAsp4MABMioxcG4Qh5Ek1CM8Cb10Z9HNMU%2BNDjkkwynVqmK1d%2BlOtoqdVE2gYiJfOhxgm8vlQEkASyKiZBgMkXBPJ4TWa3LkHM5LrvDbjf7ONlnXeBB11ioJ7XwaNiHEvBk2UjN2HzXyG1soRXJzbhi0gNB0u7ez26HSxwV6IF3jBXxF0qPqt0ezd273U0j3FdnW3MzR73AEsstJt9EZa7s-3KbjCgg5p4PaYWST6WjoPVgjPBI2TSbWRQXY9lj1KOXPUZuVOhOG-DngqP5hc8SF6-%2BhF4rRQ6UAEzOW-6SlUBpXgJ9AhTI7UzXhWHl0DdfZnbOS%2BaHmfMBZCxFmLNAEt%2B5SybuJeiP8XI62fMrDiQUoAQCdgA4EGs9YG1vsWY2KhTaeHNmgS2NBra2zfA7BgGCoFGiglAyuQcvZtX-L7IC-tuqB2DmwgaCFCTjUjPwqAgJvYOCoG4DwXgfD%2BCCKEVw7guCeDIN4PwgQQhhESMkVIdp8h-AYGqVRhRigkDKBUaooiP7niet-EkqEhgjDGBMI8UwZhnGcSsCgawNgSHbqScOxw3ELEuCYG47JlrJx5GRNO0FaoQihDCCgRZEQqALnotABjuIwCahCUOUcJoBOmlhFO0SE7hM5MRKJ618Lp1EjA1mcCIK52OjADUZ11bMSugwiusFm62gyPaWKjlXr1JbpJX20kvZyW7gpJS8Y6GuQnkPCGo8kpLLhiZMyFkZlWTRhjZefZV44x2aOGG189bjxhs-Q0r8VzuysXTZ6p5a7CHKrZSqtSTkHS-IrBqH5S7OwQawn2Jy-aYA6lQSZWsmGCShYaYRFICkCKmphWaVT441PKStEp1SKIVz-o07qzTWKtI4lxIuvEmrl23NwupkU%2BlenGdCuF2tpn3l7vMr5iyYagy0vmVZlyMyTxrI6UyDZkZQIXkveyRynKSvxlvG%2B3l5wIwPuSo%2BoVJWRQvkM85O9komAfulXGtykzvwaA9IqTzaavOvB892Gtflvn%2BY6lhIcZlgoDiy%2BFtK%2BFh2RchSOhTUWx1TmUpOlS1oYrxduAlfj4XEvzhQVWGIqWCW6dXC1LyhA2Jet1Albd4HwrZSuDlgN5U8uWSPXSY8SYaThtPWeFaOyL1sgc7GcquWb28kq4mBlrl9FNVGc1u5s3-0YQS2xCykBc08LzK0ICdBgJIOLWg0641FpXIgxWyCU3nVdVy75T4cHAB3gQohjoLZW3cJQ%2B22BHYAswTCxhvr7xiPYV1aFr7QzApGvkoNAailoqjeRJaFTVpx1A582NoyZabvlmXPObFD7FzTV0t2lEHmtHriM%2BlsDC0dRLSOMtQMXKDwEHyyGtbyPYCnnvGeVoJVdr2a2zGhyByduBgqntFzd5%2BVICh3iA7dhnyijFOKireMGtSsak5Q7Nxeyw1-BmE7YNeinYezmQCF2CyXTAUWK6IFrs01RNTsp4Pa0dUrdMKCi5oNoZphD4gT1nuxBekhZCKF22oQ5l23633%2Be1r%2Bj9nCv0Zp-e6n1-67zCNEb%2BhMbIMjrhoRIxRyjVGyI0QoqRKiZHqNCOEKIMQ4gJARDojIghiswGMSUcolQagGhppanN46YtkkceMSYXl3FhO8ZseNigo7deCVcMJEaINhpqbE7E8ToRzDhGVpEFAit-DiDkv9ASkXkiQkB0NpTMXjZxdGza8KN1NOYi0tpas0MV0C3iXp4kBnieGfmszPoLPml6kGamRHS2KQBqRrjlaKPD35TWtZwPaMirrOK4TpyW3SpXhxuHInuOeXWFwEgAR2IkC4FUHgAAKMHUNjB4AJoKpcFBJDFk6AAShRxQeTw7GFKfrjay8dqTsIKs383zQLItbo4eC9h3qf3hZ9aLo00X4VDZDUdqDWLIkgZiXS8UDLzPnYEpdslQVKWl2pWLjD%2BK3tMo7pL0Mv2kwkebfWsGKzwcU9hlDzZsOtXn2e3qvBomdUScsAxv4LB74AHkYgrdiDAOHTPDQjvZ2z5rbybzQe5z8l8zq%2BdfWCx6oXXqvvgXQ57P10uFSy5jvLtOivI2QZV8bvDDSBtHsQ9rvdHTLq3fFwqJTuaLOmdry3AjvtLdRmt8xyHlGBV1qd3Rv3TaR8I7bTK5HNu0DnIx1jnHePCfE9raT8nE-GxU5p-Tk1yW36KazZ-Vr1Ue9q9gRpl2s6GDzv5rp4W%2BnwGQJM2dhZTmd02eb%2BnjmRBFzPBc9GAM2K9chG9bze9FLA9PzdvDPBA38TPDPT9DuO7QaAXPEf1bbCOPhEvYpdFBXQ7Ig6vGDXvODTXRvElAudVVDfXdNI3TNUdT%2BLvXDG-BpfvD1Qff0YfIHIVKtLfCHAQhtejGffgv4fZBfNeJfT3ZVPjfeATOgoTN3MTS%2BOQvtbVD3MQxjQPQ1dAdwOITVOTE-VsGPZrS-BBSdFTEzB-J-RdV-AzVdJ9L-RzGqFPZBdiVBdBR9b-dw49RKXBeQ0A8A0ha9G2aAh9OAwAjA9ApAmlLA5AtA5A%2BI1qRIqXTbIRHAuLdqBLVtZLawSRJRaRNRORFQOQZgIo9LPLMo0rbRSwL4SrVbarNkExMxerVyLQ9QreMQVfbHXHfHInKjEsHfL3e%2BffanAATlp22CEU70sLxHa1GE6xcWG0WFblWHWH603XsUAyCQuFGw2Irwm32xjV-DiUhDmwW3qNSWW2iGaIoAAGoKAWAuihkOJ0lMlVF1sroi8Zdg1S9SDw0IlK9JszjyCOCW568FRE0TpNRU0GCuFUjPs%2Bp1cntL4ucISWYxkPtgRzcO4eDWw%2BC5ZuURC7dq0oYaNjJRUtkmMSTSTJDWN21ZUGdUczlJM%2Bj19BihD5xRj5D%2B0D86dWTGdTD7wn1WdrV49Ock8XYec09fCTMUCgVkiM98TAJkTWo1SEVBEK4CDgMq9gTwMy8E52DsTKDfZYTSVBMQobs4imDfwCVTdtY1Ti0fouViSJCncx8HcJ9hUXdzI6SOZvdtCeN9U3jfdEZdDg9Q97jw8Gco9XR7lz9zw49s0E97Vp05TGoFSr8lTUDQtVTc9%2Bo7SC8aU-ji8ATCDldDTsUgSptXsKCs4OpLSrsETc4kT7TmD2c2DfYC1oTXSu5qoPSyN1lySeTfSjIFD-dAzPSpCkcZDNNu10cuBMd%2BiN8hjx8%2BS%2B00YJjD94zRSzCWdkzHk79ADrDcTtw7DgEX9l1nC-Dr8zSmy-Cf9rMVZ4T90flnz-DdZAjT0QC3MwDiEICvMqEYCADcyNSaVILAJ0ikiCygsNT31btyzFjKz9SwSwNazqz6zoVXCvoWydcKUbTESwtOzQx5inkOo%2ByLzZI3Shz-s%2B5FzRzvTKTRyp9IylwFlgzujQyvdwyr4dCA974DCyAjD9zcYkyWCUyjwbCXZzzMFAE51rzQE39DMP96SHz1dCU3CXzPDvDwLADsFfzXMTZALL0wjICIjQKojPy3Dn1FToLsDYKaUVSUiyKfVnLWoUKvKhg7FM9Ci0sSjMt5FGAnIQBqgYQyAEAABtEAWEbwCAAAfQADl8AQAABdRsEAPAcxKoKK%2BAaKgAdk6G9HSukCAA)

**Reply by aweell** [reply]
Hello @annefranca, 

I don't fully understand the case active of the snapcards, but I think if this a necessity this should be discussed in another track, related to selected states there was another discussion: https://github.com/Telefonica/mistica-design/discussions/1106. We can reopen if this is the case.

Related to the main topic, the snapCard changes in the background color, we are in the same position that @yceballost explained in this comment: https://github.com/Telefonica/mistica-design/discussions/1343#discussioncomment-6438055. And we don't see how the issues listed in that comment can be solved with the new proposal.

**Reply by annefranca** [reply]
@aweell 
Acredito que seria a mesma aplicação de "ativo" da proposta dos Chips "isTransparent", certo?

**Comment by yceballost** [comment]
After reviewing the proposal together with the comments we raised in the previous messages we consider that this should be done in an extended way. 

The current cards already cover the need and, in fact, work better as they ensure good readability and usability on any color background. This proposal opens the door to possible accessibility misuse. As the intended use seems to be very specific, we believe it should be a Extended Component. If in the future we find a way to _coretize_ this, we can reevaluate it.
